### PR TITLE
Comments: scroll to most-recent during load and when switching sort setting

### LINKED ISSUE #7818: Ensure comment sidebar is scrolled to bottom on load
When ordered by "Most recent", follow up to https://github.com/outline/outline/pull/7770

### PR DESCRIPTION
**Pull request by infinite-persistence**
## Ticket
Closes #7818

## Notes
- Factored out `readyToDisplay` to use as the dependency since refs can't be one. The effect then covers both the mounting + switching cases.
    - It also ensures `document` can never be null, hence the typescript suppression.
- Due to the `50vh` bottom padding, the last comment appears in the middle of the screen. But I think this is the expected style given the padding.  If not, the scroll position can be adjusted further.
    - In addition, the lack of a scrollbar might make it look like there are less comments, but again, I think this is expected (I got the same feeling when I opened the `/` command popup for the first time).

_p/s: I skipped the pre-PR discussion part given the size of the PR. I'm totally ok with re-doing/scrapping/dropping, etc._

### CODE REVIEW

**Comment by tommoor** [comment]
Thanks! We can lose the 50vh bottom padding btw, the aim here is to make it feel a bit more like a standard chat – so it should start at the bottom and stay pinned to the bottom if you add a new comment to the document.

The big problem with this as it stands is the smooth scrolling – the user should not be able to tell that any scrolling occurred as it's quite disorienting.

**Comment by infinite-persistence** [comment]
Made some changes. Full details inside each commit message.


https://github.com/user-attachments/assets/0bc836a9-e2fb-4785-ab37-15843a6c9865

File: app/scenes/Document/components/Comments.tsx [hunk 1]
```diff
@@ -50,19 +56,62 @@ function Comments() {
...
+          scrollToBottom(); // Remain pinned to bottom on new comments
+        } else {
+          setShowJumpToRecentBtn(true);
+        }
+      }
+    }
+    prevThreadCount.current = threads.length;
+  }, [threads.length]);
 
-  if (!document || !isEditorInitialized) {
```
Comment (tommoor) [inline app/scenes/Document/components/Comments.tsx:57]: Why did you remove the `document` guard in favor of lots of non-null assertion? Seems objectively worse 🤔

File: app/scenes/Document/components/Comments.tsx [hunk 2]
```diff
@@ -132,10 +186,32 @@ const NoComments = styled(Flex)`
 `;
 
 const Wrapper = styled.div<{ $hasComments: boolean }>`
-  padding-bottom: ${(props) => (props.$hasComments ? "50vh" : "0")};
   height: ${(props) => (props.$hasComments ? "auto" : "100%")};
 `;
 
+const ScrollToRecentSize = "32px";
+
+const ScrollToRecent = styled.div`
```
Comment (tommoor) [inline app/scenes/Document/components/Comments.tsx:194]: Could you use the existing `ButtonSmall` component and just make it rounded? Could use the text `New comments` perhaps

**Review by tommoor — COMMENTED** [review]
Seems to work well! Need to figure out what's causing https://github.com/outline/outline/issues/7817 for it to feel really solid when opening

File: app/scenes/Document/components/Comments.tsx (hunk 1 above)
Comment (infinite-persistence) [reply app/scenes/Document/components/Comments.tsx:57]: The null guard was DRY-ed for re-use as an effect dependency, but Typescript for some reason can't figure out that `document` will never be null, resulting in the need to suppress the check everywhere.

Agree that it's more messy, so un-DRY-ed

File: app/scenes/Document/components/Comments.tsx (hunk 2 above)
Comment (infinite-persistence) [reply app/scenes/Document/components/Comments.tsx]: Converted.

(_and wow, the translation script seems to be able to figure out where to inject the new string in the json :+1:_ )

**Review by tommoor — APPROVED** [review]
